### PR TITLE
Pmemfile.v12.this is a branch

### PR DESCRIPTION
**Pull request by GBuella**
This change is [Reviewable](https://reviewable.io/reviews/marcinslusarz/nvml/59)

### CODE REVIEW

File: src/libsyscall_intercept/Makefile [hunk 1]
```diff
@@ -42,12 +42,11 @@ SOURCE = \
 	patcher.c \
 	intercept_util.c \
 	intercept_desc.c \
-	disasm_wrapper.c \
-	../libpmem/cpu.c
```
Comment (marcinslusarz) [inline src/libsyscall_intercept/Makefile:46]: Please also remove changes from cpu.c added in this branch.

File: src/libsyscall_intercept/intercept_desc.c [hunk 2]
```diff
@@ -152,7 +151,7 @@ has_jump(const struct intercept_desc *desc, unsigned char *addr)
 	if (addr >= desc->text_start && addr <= desc->text_end) {
 		uint64_t offset = (uint64_t)(addr - desc->text_start);
 
-		return util_isset(desc->jump_table, offset);
+		return desc->jump_table[offset / 8] & (1 << (offset % 8));
```
Comment (marcinslusarz) [inline src/libsyscall_intercept/intercept_desc.c]: Can't we just copy util_isset and util_setbit to syscall_intercept?

File: src/libsyscall_intercept/patcher.c [hunk 3]
```diff
@@ -128,8 +128,6 @@ create_absolute_jump(unsigned char *from, void *to)
 	from[11] = d[5];
 	from[12] = d[6];
 	from[13] = d[7];
-
-	COMPILE_ERROR_ON(TRAMPOLINE_SIZE != 14);
```
Comment (marcinslusarz) [inline src/libsyscall_intercept/patcher.c:132]: Again, can't we just copy this macro?

**Review by marcinslusarz — CHANGES_REQUESTED** [review]

**Comment by GBuella** [comment]
Review status: 0 of 10 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/libsyscall_intercept/intercept_desc.c, line 154 at r1](https://reviewable.io:443/reviews/marcinslusarz/nvml/59#-KbtCmYLsg5FhmXIa4ne-r1-154:-KbtU6PcpXc82nlXuZAB:b-qybfbo) ([raw file](https://github.com/marcinslusarz/nvml/blob/45a902b5a75abb162597136521255badf1b9af70/src/libsyscall_intercept/intercept_desc.c#L154)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Can't we just copy util_isset and util_setbit to syscall_intercept? 
</blockquote></details>

I did, sort of.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/nvml/59)*

**Comment by GBuella** [comment]
> Please also remove changes from cpu.c added in this branch.
Done

---

Review status: 0 of 10 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/marcinslusarz/nvml/59#-:-KbtULSDz96D7VNQp53B:bj3p6gh)*

File: src/libsyscall_intercept/Makefile (hunk 1 above)
Comment (GBuella) [reply src/libsyscall_intercept/Makefile:46]: Done.

**Pull request merged**